tests/ci/lambda/src/bin/purge-stale-builds.rs

```diff
@@ -119,6 +124,58 @@ async fn handle(_event: LambdaEvent<Value>) -> Result<(), Error> {
         .to_string()
     );
 
+    // Filters for aws-lc-ec2-test-framework specific hosts.
+    let ec2_client = aws_sdk_ec2::Client::new(&config);
+    let ec2_framework_filters = vec![
+        Filter::builder()
+            .name("instance-state-name")
+            .values("running")
+            .build(),
+        Filter::builder()
+            .name("instance.group-name")
+            .values("codebuild_ec2_sg")
+            .build(),
+    ];
+
+    let ec2_describe_response = ec2_client
+        .describe_instances()
+        .set_filters(Some(ec2_framework_filters))
+        .send()
+        .await;
+
+    if ec2_describe_response.is_err() {
+        log::info!("No IAM Permissions to DescribeInstances");
+        return Ok(());
+    }
+
+    let mut ec2_hanging_instances: Vec<String> = vec![];
+    for reservation in ec2_describe_response.unwrap().reservations() {
+        log::info!("Checking Instance {:?}", reservation.instances());
+        for instance in reservation.instances() {
+            // Prune ec2 instances hanging more than 120 minutes.
```

Comment:
Can this also handle cleaning up duplicate instances in the case of second push to the same PR creating a new job and the results of the first job being unused?

Reply:
To add to this, we should probably use tags on the instance to indicate the metadata about what pull request its related to, what the commit head is when you launch it with run-instances, and tag it with something to indicate that it is part of the ec2-test-framework. Then your filter here can be to look for that indicator tag, and then you can use the other tags with the PR number, head commit etc to determine if an instance should be nuked cause its an old push.

Reply:
Thanks for the feedback Sean! Yeah it would be a bit more to support this, but doable. 
I was planning on adding the `CODEBUILD_SOURCE_VERSION`(which contains the commit hash) [to the ec2 tags](https://github.com/aws/aws-lc/blob/main/tests/ci/run_ec2_test_framework.sh#L37), then [reuse a bit of your PR commit logic above](https://github.com/aws/aws-lc/blob/main/tests/ci/lambda/src/bin/purge-stale-builds.rs#L68-L73) to terminate the right instances.
I guess that coincides with your proposal, do let me know if there's any part of the puzzle I'm missing though.

Reply:
I got things working and the ec2 instances are now pruned at the same pace as our Codebuild jobs.